assimilation_code/programs/readme.rst

```diff
@@ -0,0 +1,180 @@
+Programs included with DART
+===========================
+
+This list of programs is separated into groups which have similar functionality.
+Within each group they are sorted  by the order
+in which they might be used and/or by how widely they are used.
+
+Setting Up Experiments
+-----------------------------------
+
+In many cases, you won't need to use any programs in Setting Up Experiments
+except for ``preprocess``, because you're using an existing model interface 
+and have the observation sequence files.
+In that case, you the programs you're looking for are probably in `Assimilation Programs`_.
+ 
+:doc:`preprocess <preprocess/preprocess>`
+   Program to insert observation specific code into DART before filter or perfect_model_obs is compiled.
+ 
+:doc:`fill_inflation_restart <fill_inflation_restart/fill_inflation_restart>`
+   Create inflation restart files with constant values taken from ``fill_inflation_restart_nml``.
+
+:doc:`obs_impact_tool <obs_impact_tool/obs_impact_tool>`
+   Construct a table that is read by filter at run-time to localize the
+   impact of sets of observation types on sets of state vector quantities.
+ 
+:doc:`model_mod_check <model_mod_check/model_mod_check>` 
+  Program to test some of the more fundamental routines in any ``model_mod``, especially a for a new model.
+
+:doc:`perturb_single_instance <perturb_single_instance/perturb_single_instance>`
+   Generate an ensemble of perturbed ensemble member restart files.
+   (Alternatively, you might perturb the model state using ``model_nml`` variables).
+ 
+:doc:`gen_sampling_err_table <gen_sampling_err_table/gen_sampling_err_table>`
+   Computes a table of values needed to apply Sampling Error Correction (SEC),
+   which corrects covariances based on small sample size statistics.
+ 
+Creating Observation Sequence Files
+-----------------------------------
+
+obs_utils/create_obs_grid
+   Create a set of observations located on a regular grid.  Obs have no data values.
+
+:doc:`create_fixed_network_seq <create_fixed_network_seq/create_fixed_network_seq>` 
+   Reads observation sequence file information from standard input 
+   and replicates it multiple times in a second observation sequence file, at user specified dates. 
+ 
+obs_utils/obs_timejitter
+   Randomly perturb the times of the observations in a (usually) set_def.out file.
+   Writes the results to (usually) obs_seq.in.
+
+:doc:`create_obs_sequence <create_obs_sequence/create_obs_sequence>`
```

Comment:
suggested order in this section:  create_obs_sequence, create_fixed_network_seq, create_obs_grid obs_timejitter

Reply:
My feeling is that when I go to a website, if I know what I'm looking for, I search for it and go directly there.
If I don't know, I want the pieces to be orgainized in some order that I can see or understand.
Ordering by how often a program is used or how frequently the documentation is referenced 
doesn't help me much.
Ordering by which programs create output which could be used as inputs by others makes sense to me.
I could be an outlier.

Reply:
i agree. the order these programs are used is normally create_obs_sequence first, and output of that is the input to create_fixed_network_seq. the other two are infrequently used. so my suggested order in this section fits your criteria, does it not?

Reply:
My mistake; I was led astray by the mention of obs_seq.in in create_obs_grid, 
but not in the create_obs_sequence.rst.  Now that I've read all the comments
it's clear that create_obs_sequence should be first.

Reply:
Also, it seems that create_fixed_network_seq creates a time sequence from whatever array of obs locations
is fed to it, so create_obs_grid could or would create input for that.  But I'm gettig the impression 
that create_obs_grid was created just to create an obs_seq.in for perfect_model_obs.